src/empty-state.tsx

```diff
@@ -0,0 +1,161 @@
+import * as React from 'react';
+import Box from './box';
+import {Boxed} from './boxed';
+import {ButtonPrimary, ButtonSecondary, ButtonLink} from './button';
+import {useScreenSize, useTheme} from './hooks';
+import Stack from './stack';
+import {Text2, Text4, Text6} from './text';
+import {createUseStyles} from './jss';
+import Inline from './inline';
+import {getPrefixedDataAttributes} from './utils/dom';
+
+import type {DataAttributes} from './utils/types';
+
+const useStyles = createUseStyles((theme) => ({
+    container: {
+        [theme.mq.tabletOrSmaller]: {
+            maxWidth: 368,
+        },
+    },
+
+    desktopContainer: {
+        display: 'flex',
+        justifyContent: 'spaceBetween',
+    },
+    desktopContent: {
+        [theme.mq.desktopOrBigger]: {
+            width: '50%',
+        },
+    },
+    desktopImage: {
+        backgroundImage: ({largeImageUrl}) => `url(${largeImageUrl})`,
+        backgroundPosition: 'bottom right',
+        backgroundRepeat: 'no-repeat',
+        backgroundSize: 'contain',
+        flex: 1,
+    },
+    largeImage: {
+        width: '100%',
+        aspectRatio: '16/9',
+    },
+    smallImage: {
+        height: 128,
+        [theme.mq.tabletOrSmaller]: {
+            height: 112,
+        },
+    },
+}));
+
+interface BaseProps {
+    title: string;
+    button?: React.ReactElement<typeof ButtonPrimary> | React.ReactElement<typeof ButtonSecondary>;
+    buttonLink?: React.ReactElement<typeof ButtonLink>;
```

Comment:
Same here. Maybe we should have an eslint rule to avoid using React.ReactElement with only one generic param